packages/data-access-plugin/src/spaces/dynamic-schema.ts

```diff
@@ -40,60 +39,69 @@ function sanitize(name: string) {
         .trim();
 }
 
+function makeEndpoint(endpoint: string) {
+    return `${endpoint}(join: [String], query: String, size: Int, from: Int, sort: String): [${endpoint}!]!`;
+}
+
 function createTypings(configs: DataAccessConfig[]) {
-    const results: string[] = ['scalar JSON', 'scalar DateTime', Usertype, GeoType];
-    const queryEndpoints: string[] = configs.map(config => config.space_endpoint);
-    // create individual types
-    configs.forEach(config => {
-        results.push(`type ${config.space_endpoint} { ${collectAllowedFields(config, queryEndpoints)} }`);
-    });
+    const results: string[] = ['scalar JSON', 'scalar DateTime', Usertype];
+    const endpointList: string = configs.map((config) =>  makeEndpoint(config.space_endpoint)).join('\n    ');
+    const filteredDataTypes = configs.map(filterDataTypes);
+    const myTypes = DataType.mergeGraphQLDataTypes(filteredDataTypes, endpointList);
     // create query type
     results.push(
+        myTypes,
         ` type Query {
-                ${queryEndpoints
-                    .map(endpoint => `${endpoint}(join: [String], query: String, size: Int, from: Int, sort: String): [${endpoint}!]!`)
-                    .join('\n    ')}
+                ${endpointList}
             }
         `
     );
 
     return results;
 }
 
-function pick(obj: object = {}, keys: string[]) {
-    const results = {};
-    keys.forEach(key => {
-        if (obj[key] !== null) results[key] = obj[key];
+function hasKey(values: string[], field:string) {
+    const results = values.filter((value) => {
+        return (value === field || value.match(new RegExp(`${field}\\.`)));
```

Comment:
This regex needs to have a starts with character

```suggestion
        return (value === field || value.match(new RegExp(`^${field}\\.`)));
```